Mathlib/NumberTheory/GaussSum.lean

```diff
@@ -97,8 +97,15 @@ lemma gaussSum_mul {R : Type u} [CommRing R] [Fintype R] {R' : Type v} [CommRing
     · exact fun a _ ↦ by rw [add_sub_cancel_right, add_comm]
   rw [sum_congr rfl fun x _ ↦ sum_eq x, sum_comm]
 
--- In the following, we need `R` to be a finite field and `R'` to be a domain.
-variable {R : Type u} [Field R] [Fintype R] {R' : Type v} [CommRing R'] [IsDomain R']
+-- In the following, we need `R` to be a finite field.
+variable {R : Type u} [Field R] [Fintype R] {R' : Type v} [CommRing R']
```

Comment:
How annoying would be to replace `Fintype` with `Finite`? Don't worry if it is a big refactor.

Reply:
Basically impossible; you cannot talk about sums over a finite type unless you have a `Fintype` instance on it (`Finset.univ` requires `Fintype`).